File: app/src/main/java/temple/edu/yelp_randomizer/restaraunts/RandomRestaurantsAdapter.java
package temple.edu.yelp_randomizer.restaraunts;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;
import temple.edu.yelp_randomizer.View.RestaurantView;
import temple.edu.yelp_randomizer.models.RestaurantModel;


import java.util.ArrayList;

public class RandomRestaurantsAdapter extends BaseAdapter {
    private ArrayList<RestaurantModel> restaurants;
    Context context;

    public RandomRestaurantsAdapter(Context context, ArrayList<RestaurantModel> restaurants){
        this.restaurants = restaurants;
        this.context = context;
    }
    @Override
    public int getCount() {
        return restaurants.size();
    }

    @Override
    public Object getItem(int i) {
        return restaurants.get(i);
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        RestaurantView restaurantView;

        if (view != null){
            restaurantView = (RestaurantView) view;
        }else{

            restaurantView = new RestaurantView(context, null);

        }
        String text = restaurants.get(i).getName() + "\n" +
                      convertRating(restaurants.get(i).getRating()) + "\n\n" +
                      restaurants.get(i).getPhone() + "\n\n" +
                       restaurants.get(i).getLocation();
        restaurantView.setTextView(text);
        restaurantView.setImageView(restaurants.get(i).getImage());

        return restaurantView;
    }

    private String convertRating(int rating){
        String rating_stars = "";
        for (int i = 0; i<= rating-1; i++){
            rating_stars += '⭐';
        }
        return rating_stars;
    }


}

File: app/src/main/java/temple/edu/random/restaraunts/RandomRestaurantsAdapter.kt
package temple.edu.random.restaraunts

import android.content.Context
import android.widget.BaseAdapter
import android.view.ViewGroup
import android.view.View
import temple.edu.random.models.RestaurantModel
import temple.edu.random.View.RestaurantView
import java.util.ArrayList

class RandomRestaurantsAdapter(var context: Context, private val restaurants: ArrayList<RestaurantModel>) : BaseAdapter() {
    override fun getCount(): Int {
        return restaurants.size
    }

    override fun getItem(i: Int): Any {
        return restaurants[i]
    }

    override fun getItemId(i: Int): Long {
        return 0
    }

    override fun getView(i: Int, view: View, viewGroup: ViewGroup): View {
        val restaurantView: RestaurantView
        restaurantView = if (view != null) {
            view as RestaurantView
        } else {
            RestaurantView(context, null)
        }
        val text = """
            ${restaurants[i].name}
            ${convertRating(restaurants[i].rating)}
            
            ${restaurants[i].phone}
            
            ${restaurants[i].location}
            """.trimIndent()
        restaurantView.setTextView(text)
        restaurantView.setImageView(restaurants[i].image)
        return restaurantView
    }

    private fun convertRating(rating: Int): String {
        var rating_stars = ""
        for (i in 0..rating - 1) {
            rating_stars += '⭐'
        }
        return rating_stars
    }
}